### PR TITLE
platform upgrader inheritance bug

### PR DESCRIPTION
**Pull request by oliver-sanders**
Sorry, one more spotted in release testing.

Inheritance wasn't playing nicely with the platform upgrader. Perhaps it was this one - https://github.com/cylc/cylc-flow/issues/3812

I'm pretty sure we've hit this before and there was an issue, can't seem to find it (open or closed)?

Example workflow:

```ini
[scheduling]
    [[graph]]
        R1 = a

[runtime]
    [[A]]
        [[[job]]]
            batch system = background
    [[a]]
        inherit = A
```

Example workflow log contents:

```
2021-03-25T13:12:34Z ERROR - [jobs-submit cmd] (platform not defined)
	[jobs-submit ret_code] 1
	[jobs-submit err]
	A mixture of Cylc 7 (host) and Cylc 8 (platform) logic should not be used. In this
	case the task "unknown task" has the following settings which are not compatible:
	 * platform = localhost AND [job]batch system = background
```

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (bug was never released)
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

File: tests/functional/platforms/02-host-to-platform-upgrade.t [hunk 1]
```diff
@@ -30,14 +30,16 @@ run_fail "${TEST_NAME_BASE}-validate-fail" \
...
     cylc validate "${SUITE_NAME}" \
         -s "CYLC_TEST_HOST='${CYLC_TEST_HOST}'" \
         -s CYLC_TEST_HOST_FQDN="'$(ssh "$CYLC_TEST_HOST" hostname -f)'"
 
 # Check that the cfgspec/suite.py has issued a warning about upgrades.
-grep_ok "\[upgradeable_cylc7_settings\]\[remote\]host = ${CYLC_TEST_HOST}"\
-    "${TEST_NAME_BASE}-validate.stderr"
+grep_ok "\[t1\]\[remote\]host = ${CYLC_TEST_HOST}"\
+    "${TEST_NAME_BASE}-run.stderr"
+grep_ok "\[t2\]\[remote\]host = ${CYLC_TEST_HOST}"\
```
Comment (datamel) [inline tests/functional/platforms/02-host-to-platform-upgrade.t:41]: I have run this test locally. Log is reporting:  `"\[T2\]\[remote\]host = ${CYLC_TEST_HOST}"\`, rather than the `t1` in the test.

Comment (wxtim) [reply tests/functional/platforms/02-host-to-platform-upgrade.t:41]: I can confirm that I get this error too.

Comment (oliver-sanders) [reply tests/functional/platforms/02-host-to-platform-upgrade.t]: Had a dig, because of the way these warnings are raised it cannot tell tasks from families so I've updated it to `T2`.

**Review by wxtim — APPROVED** [review]
🦛 Happy

**Comment by datamel** [comment]
`tests/f/deprecations/01-cylc8-basic.t ` is failing actions but passing locally for me.

**Comment by datamel** [comment]
> `tests/f/deprecations/01-cylc8-basic.t ` is failing actions but passing locally for me.

Looks like culprit possibly could be whitespace difference: `...global.cylc, but support...`  vs `...global.cylc,but support...`

File: cylc/flow/cfgspec/suite.py [hunk 2]
```diff
@@ -1522,12 +1522,12 @@ def upg(cfg, descr):
...
-                    "setting is deprecated; use"
-                    "'global.cylc[platforms][<platform name>]{job_setting}' "
+                    f"* (8.0.0) '[runtime][{task}]{job_setting}' - this "
+                    "setting is deprecated; use "
+                    f"'global.cylc[platforms][<platform name>]{job_setting}' "
                     "instead. "
                     "Currently, this item will override the corresponding "
-                    "item in global.cylc,"
+                    "item in global.cylc, "
                     "but support for this will be removed in Cylc 9."
```
Comment (oliver-sanders) [inline cylc/flow/cfgspec/suite.py:1531]: Unrelated fix to a warning.

**Review by datamel — APPROVED** [review]
GH actions now passing. Thanks Oliver.

**Review by hjoliver — APPROVED** [review]
